### PR TITLE
Less string comparison in ExpressionActions.

### PR DESCRIPTION
**Pull request by KochetovNicolai**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Category (leave one):
- Improvement

Short description (up to few sentences):
Decreased the number of string comparisons while executing actions on block. Split long_accurate_number_comparison test.

### CODE REVIEW

File: dbms/src/Core/Settings.h [hunk 1]
```diff
@@ -326,7 +326,7 @@ struct Settings : public SettingsCollection<Settings>
     M(SettingBool, external_table_functions_use_nulls, true, "If it is set to true, external table functions will implicitly use Nullable type if needed. Otherwise NULLs will be substituted with default values. Currently supported only for 'mysql' table function.") \
     M(SettingBool, allow_experimental_data_skipping_indices, false, "If it is set to true, data skipping indices can be used in CREATE TABLE/ALTER TABLE queries.") \
     \
-    M(SettingBool, experimental_use_processors, false, "Use processors pipeline.") \
+    M(SettingBool, experimental_use_processors, true, "Use processors pipeline.") \
```
Comment (amosbird) [inline dbms/src/Core/Settings.h]: hmm, is this intentional and indicating a production ready status of the processor pipeline?

Comment (amosbird) [reply dbms/src/Core/Settings.h]: oh, that reminds me of toggling the flag on for the sso map too. thanks

Comment (alexey-milovidov) [reply dbms/src/Core/Settings.h]: It is not necessarily if your SSO map is strictly better and well tested.

File: dbms/src/Interpreters/ExpressionActions.cpp [hunk 2]
```diff
@@ -329,16 +382,51 @@ void ExpressionAction::prepare(Block & sample_block, const Settings & settings)
 }
 
 
-void ExpressionAction::execute(Block & block, bool dry_run) const
+template <bool execute_on_block>
+void ExpressionAction::execute(
+    Block & block,
```
Comment (akuzm) [inline dbms/src/Interpreters/ExpressionActions.cpp]: Putting every other line in a function under an `#ifdef` (oh sorry it's called `if constexpr` now) is a time-tested, respectable way to make it totally unreadable. If you have to perform the same actions on two kinds of data structures, you could make them support the same interface, either by changing them or by writing a wrapper.

File: dbms/src/Processors/Executors/PipelineExecutor.cpp [hunk 3]
```diff
@@ -577,32 +577,45 @@ void PipelineExecutor::executeImpl(size_t num_threads)
     for (size_t i = 0; i < num_threads; ++i)
         executor_contexts.emplace_back(std::make_unique<ExecutorContext>());
 
+    auto thread_group = CurrentThread::getGroup();
+
```
Comment (akuzm) [inline dbms/src/Processors/Executors/PipelineExecutor.cpp]: This part looks unrelated, should it be split to a separate commit?

Comment (KochetovNicolai) [reply dbms/src/Processors/Executors/PipelineExecutor.cpp]: Oh, I have forgotten about that small change.
Can we keep it here?

**Comment by alexey-milovidov** [comment]
@KochetovNicolai Let's pull master and we'll continue reviewing.

**Comment by KochetovNicolai** [comment]
Close for now.
